[Statspadders] iP

Statspadders opened the pull request:
# Welcome to Duke
>The famous Ryochi once said that in your life, you need to have Duke to store all your tasks.

You can download it [here](https://github.com/Statspadders/ip)

Benefits of Using Duke
+ Free
+ Pay to win
+ Fastest program

1. Best
2. LOL
3. Pro

```ruby
require 'redcarpet'
markdown = Redcarpet.new("Hello World!")
puts markdown.to_html
```

- [X] Hello world
- [ ] Testing
- [ ]  What

@octocat :+1: This PR looks great - it's ready to merge! :shipit:

`Hello world` `Java` `Donezo`

_testing_

Commit:
Add Gradle support

Statspadders committed:
Level-1

Statspadders committed:
Level-2

Statspadders committed:
Level-3

Statspadders committed:
Level -4 ToDo, Event, Deadline

Statspadders committed:
Level-5 Handle Errors

Statspadders committed:
Level-6 Remove

Statspadders committed:
Level-7. Save

Statspadders committed:
branch-Level-8 Dates and Times

Statspadders committed:
Merge branch 'branch-Level-7'

Statspadders committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

Statspadders committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Statspadders committed:
A-MoreOOP

Statspadders committed:
A-Packages
Divide classes into packages. Put all classes into duke package, organize commands in duke.command

Statspadders committed:
A-JUnit

Statspadders committed:
A-JavaDoc

Statspadders committed:
A-CodingStandard

Make minor coding style formatting to Parser.java

Statspadders committed:
branch-Level-9
Added feature find to duke.

Statspadders committed:
Merge branch 'branch-A-JavaDoc'

Statspadders committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java

Statspadders committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/command/Command.java

linustws commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,42 @@
+import duke.*;
```
Perhaps import each class explicitly, as per the Java coding standard. I noticed the same issue in several other places too.

linustws commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,75 @@
+     * Write a new line to the file
+     * @param t = task
+     */
+    public void write(Task t){
```
Perhaps 'save' would be a better function name than 'write'? because it goes with 'load'. Or maybe changing 'load' to 'read', such that it matches better.

linustws commented on src/main/java/duke/command/CommandNotFound.java:
```diff
@@ -0,0 +1,27 @@
+ * CommandNotFound - when user enters a Command and Duke
+ *         does not understand the command.
+ */
+public class CommandNotFound extends Command {
```
Perhaps consider changing the name of this class to be InvalidCommand or UnknownCommand, to be consistent with the naming of the rest of the command classes.

linustws commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,69 @@
+        String value = br.readLine();
+        return value;
+    }
+    public void showMarkSucess(Task value) {
```
Consider rectifying the spelling/gramatical error here to showMarkSuccessful or maybe just change it to showMarked, makes it less lengthy yet concise. Also, isn't there supposed to be an unmark command too?

linustws reviewed (commented):
LGTM, almost. Just a few nits to fix. 👍

Statspadders commented:
> LGTM, almost. Just a few nits to fix. 👍

Thank you for your feedback !

Statspadders committed:
Level-10
Created a new GUI for Duke.
Some minor bug fixes.

Statspadders committed:
A-Assertions

Added Assert to aid with error checking or testing of the application

Add "Assert" to the command variable to check whether it is null at the end of the parsing. If it is null, it will reflect with the message that command cannot  be null

Statspadders committed:
Merge pull request #1 from Statspadders/branch-A-Assertions

A-Assertions

Statspadders committed:
branch-A-CodeQuality

Statspadders committed:
Merge pull request #2 from Statspadders/branch-A-CodeQuality

branch-A-CodeQuality

Statspadders committed:
Add archive feature

Duke does not have an archive feature

To enhance the user experience, as time goes by, the tasklist might be filled with a lot of tasks and it will be messy and hard for the user to keep track of his tasks.

Duke now has an archive feature, which can be accessed by typing "Archive" into the input box. Archive will delete all of the tasks in the tasklist as well as the data in data/duke.txt. The deleted data will be archived in a new file named "current datetime eg. 021120232053" and saved in the directory "data".

Statspadders committed:
Add archive feature

Duke does not have an archive feature

To enhance the user experience, as time goes by, the tasklist might be filled with a lot of tasks and it will be messy and hard for the user to keep track of his tasks.

Duke now has an archive feature, which can be accessed by typing "Archive" into the input box. Archive will delete all of the tasks in the tasklist as well as the data in data/duke.txt. The deleted data will be archived in a new file named "current datetime eg. 021120232053" and saved in the directory "data".

Statspadders committed:
A-MoreErrorHandling

Statspadders committed:
A-MoreErrorHandling

Statspadders committed:
Update README.md

Statspadders committed:
Update Readme.md

Statspadders committed:
Update Readme.md

Statspadders committed:
Update README.md

Statspadders committed:
Minor updates

Statspadders committed:
Update README.md

Statspadders committed:
Update README.md

Statspadders committed:
UPDATE build.gradle

Change Javafx version to 11. To fix error when running on linux

Statspadders committed:
UPDATE build.gradle
